lib/types/array.js

```diff
@@ -0,0 +1,157 @@
+// Copyright IBM Corp. 2016. All Rights Reserved.
+// Node module: strong-remoting
+// This file is licensed under the Artistic License 2.0.
+// License text available at https://opensource.org/licenses/Artistic-2.0
+
+'use strict';
+
+var assert = require('assert');
+var debug = require('debug')('strong-remoting:http-coercion');
+var escapeRegex = require('escape-string-regexp');
+var g = require('strong-globalize')();
+
+module.exports = ArrayConverter;
+
+function ArrayConverter(itemType) {
+  this._itemType = itemType;
+}
+
+ArrayConverter.prototype.fromTypedValue = function(ctx, value) {
+  if (value === undefined || value === null)
+    return { value: value };
+
+  if (!Array.isArray(value))
+    return { error: notAnArrayError() };
+
+  var items = new Array(value.length);
+  var itemResult;
+  var itemConverter = ctx.typeRegistry.getConverter(this._itemType);
+
+  for (var ix in value) {
+    itemResult = itemConverter.fromTypedValue(ctx, value[ix]);
+    itemResult = validateConverterResult(itemResult);
+    debug('typed item result: %j -> %j as %s',
+      value[ix], itemResult, this._itemType);
+
+    if (itemResult.error)
+      return itemResult;
+    items[ix] = itemResult.value;
+  }
+  return { value: items };
+};
+
+ArrayConverter.prototype.fromSloppyValue = function(ctx, value) {
+  if (value === undefined || value === '') {
+    // undefined was chosen so that it plays well with ES6 default parameters.
+    return { value: undefined };
+  }
+
+  if (value === null || value === 'null') {
+    return { value: null };
+  }
+
+  return this._fromTypedValueString(ctx, value) ||
+    this._fromDelimitedString(ctx, value) ||
+    this._fromSloppyData(ctx, value);
+};
+
+ArrayConverter.prototype._fromTypedValueString = function(ctx, value) {
+  var looksLikeJsonArray = typeof value === 'string' &&
+    value[0] === '[' && value[value.length - 1] === ']';
+
+  if (!looksLikeJsonArray)
+    return null;
+
+  // If it looks like a JSON array, try to parse it.
+  try {
+    var result = JSON.parse(value);
+    debug('parsed %j as JSON: %j', value, result);
+    return this.fromTypedValue(ctx, result);
+  } catch (ex) {
+    debug('Cannot parse array value %j. %s', value, ex);
```

Comment:
```
$ node
> console.log('%s', new Error('asd'))
Error: asd
```

In this particular case, I think the stack trace is not that useful. If you think differently, then I can change the last argument to `err.stack || err`.